src/main/java/com/icthh/xm/ms/entity/service/ElasticsearchIndexService.java

```diff
@@ -206,8 +226,13 @@ private Long reindexXmEntity() {
     }
 
     private long reindexXmEntity(@Nullable Specification<XmEntity> spec) {
+        return reindexXmEntity(spec, null);
+    }
+
+    private long reindexXmEntity(@Nullable Specification<XmEntity> spec,  Integer startFrom) {
```

Comment:
better to add normal Pagable argument? with sorting and normal pagable?

Reply:
not sure about that.
A "reindexXmEntity()" method has loop from startFrom to last page and on each iteration it builds its own pageable object. So, if i pass Pageable i'll need to 
1) convert pageable sorting to Specification
2) get startFrom and page number and pass it to the loop in "reindexXmEntity()", which level all Pageable benefits.

Maybe i did not get ur point. In this case pls elaborate

Reply:
maybe instead of that worth to add Specification as a parameter?

Reply:
in line 251 you convert startFrom to Pagable
i see no reason to pass startFrom instead Pagable

Reply:
i do no convert startFrom into Pageable